include/rmm/mr/device/tracking_resource_adaptor.hpp

```diff
@@ -179,8 +224,17 @@ class tracking_resource_adaptor final : public device_memory_resource {
     {
       write_lock_t lock(mtx_);
       allocations_.emplace(p, allocation_info{bytes, capture_stacks_});
+
+      // Increment the allocation_count_ while we have the lock
+      allocation_counts_.current_bytes += bytes;
+      allocation_counts_.current_count += 1;
+      allocation_counts_.peak_bytes =
+        std::max(allocation_counts_.current_bytes, allocation_counts_.peak_bytes);
+      allocation_counts_.peak_count =
+        std::max(allocation_counts_.current_count, allocation_counts_.peak_count);
+      allocation_counts_.total_bytes += bytes;
+      allocation_counts_.total_count += 1;
```

Comment:
This should be encapsulated in a member function of `allocation_counts`.

Reply:
Moved to a member function. Resolving.